src/Sentry.AspNetCore/SentryMiddleware.cs

```diff
@@ -99,10 +98,13 @@ public async Task InvokeAsync(HttpContext context, RequestDelegate next)
                 // sent to Sentry. This avoid the cost on error-free requests.
                 // In case of event, all data made available through the HTTP Context at the time of the
                 // event creation will be sent to Sentry
-                scope.OnEvaluating += (_, _) =>
+
+                // Important: The scope that the event is attached to is not necessarily the same one that is active
+                // when the event fires.  Use `activeScope`, not `scope` or `hub`.
+                scope.OnEvaluating += (_, activeScope) =>
                 {
-                    SyncOptionsScope(hub);
-                    PopulateScope(context, scope);
+                    SyncOptionsScope(activeScope);
+                    PopulateScope(context, activeScope);
```

Comment:
This was required so that we are using the scope that the event fired on - which isn't necessarily the same one it's attached to (after we copy over the event handlers in the `Clone` method below).